fix: incorrect reduction of subtraction from zero (#88)

jasminexie opened the pull request:
This PR should fix #88, where a division expression subtracted from zero is incorrectly reduced. Apparently `reducer.js` does not reduce division expressions if the denominator cannot be reduced to a value, so similar multiplication tests pass while division did not.

evilebottnawi commented on src/__tests__/index.js:
```diff
@@ -360,6 +360,13 @@ test(
+test(
+  'should reduce substracted division expression from zero',
+  testValue,
+  'calc( 0 - (100vw - 10px) / 2 )',
```
Can you add tests when `0` is other value

jasminexie commented:
The changes I made affected only the '0 - something => -something' case, so correct me if I'm wrong please but I don't see a point of adding tests when 0 is another value. Tests like `value - unreducable division expression` should already be covered by `value - expression` and `division expression` tests. I have add another one if you think it hasn't been sufficiently tested.

The '0 - something => -something' case currently has the following tests:

* 0 - value
* 0 - calc expression
* 0 - unreducable division expression (in this PR)

I've added the cases:

* 0 - unreducable addition/subtraction expression
* 0 - unreducable multiplication expression

To test the actual code I changed, these tests should not be reducable, and they should preserve the node. One example is `calc( 0 - (100vw - 10px) * 2px )`. If any of these expressions are reducable, they would fall back to the `0 - value` state, which already has a test case. I've edited the PR, please tell me what you think.

evilebottnawi commented:
/cc @ben-eb can you look on this?

ben-eb commented:
I don't have much to add really, I wasn't involved with the more recent refactoring. Thanks for the contribution, @jasminexie 😃

evilebottnawi commented:
Thanks
